[BUG] npm unpublish fails if scope is involved

### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

The following command fails if .npmrc contains @someScope:registry=https://some.npm.registry/

`npm unpublish @someScope/somePackage@0.1.2-beta3`

The command works though if .npmrc contains registry=https://some.npm.registry/

The weird thing is `npm publish` for the same package works if .npmrc contains @someScope:registry=https://some.npm.registry/



### Expected Behavior

The `npm unpublish` command should work like the `npm publish` command if package scope is defined.

### Steps To Reproduce

1. In any NPM 8.19.2 environment with NodeJS 16.17.1 on Windows 11
2. With some custom package and custom registry **that is not**  https://registry.npmjs.org with .npmrc containing @someScope:registry=https://some.npm.registry/
3. Run `npm unpublish @someScope/somePackage@0.1.2-beta3`
4. See the following error:

npm ERR! 404 Not Found - GET https://registry.npmjs.org/@someScope%2fsomePackage - Not found
npm ERR! 404
npm ERR! 404  '@someScope/somePackage' is not in this registry.

This is correct, though the package clearly is in the custom registry. And switching from @someScope:registry=https://some.npm.registry/ to registry=https://some.npm.registry/ makes unpublish work again.

### Environment

- npm: 8.19.2
- Node.js: 16.17.1
- OS Name: Windows 11
- System Model Name: Irrelevant
- npm config:
```ini
@someScope:registry = "https://some.npm.registry/"
//some.npm.registry/:_authToken = (protected)
```

--- a/lib/commands/unpublish.js
+++ b/lib/commands/unpublish.js
@@ -26,7 +26,10 @@ class Unpublish extends BaseCommand {
 
   async getKeysOfVersions (name, opts) {
     const pkgUri = npa(name).escapedName
-    const json = await npmFetch.json(`${pkgUri}?write=true`, opts)
+    const json = await npmFetch.json(`${pkgUri}?write=true`, {
+      ...opts,
+      spec: name,
+    })
     return Object.keys(json.versions)
   }
 

--- a/test/fixtures/mock-npm.js
+++ b/test/fixtures/mock-npm.js
@@ -206,7 +206,7 @@ const setupMockNpm = async (t, {
         acc.env[`process.env."npm_config_${key}"`] = value
       } else {
         const values = [].concat(value)
-        acc.argv.push(...values.flatMap(v => [`--${key}`, v.toString()]))
+        acc.argv.push(...values.flatMap(v => `--${key}=${v.toString()}`))
       }
       acc.config[key] = value
       return acc

--- a/test/lib/commands/unpublish.js
+++ b/test/lib/commands/unpublish.js
@@ -27,7 +27,7 @@ t.test('no args --force success', async t => {
   })
   const manifest = registry.manifest({ name: pkg })
   await registry.package({ manifest, query: { write: true } })
-  registry.nock.delete(`/${pkg}/-rev/${manifest._rev}`).reply(201)
+  registry.unpublish({ manifest })
   await npm.exec('unpublish', [])
   t.equal(joinedOutput(), '- test-package@1.0.0')
 })
@@ -148,7 +148,7 @@ t.test('no version found in package.json', async t => {
   })
   const manifest = registry.manifest({ name: pkg })
   await registry.package({ manifest, query: { write: true } })
-  registry.nock.delete(`/${pkg}/-rev/${manifest._rev}`).reply(201)
+  registry.unpublish({ manifest })
 
   await npm.exec('unpublish', [])
   t.equal(joinedOutput(), '- test-package')
@@ -168,7 +168,7 @@ t.test('unpublish <pkg> --force no version set', async t => {
   })
   const manifest = registry.manifest({ name: pkg })
   await registry.package({ manifest, query: { write: true }, times: 2 })
-  registry.nock.delete(`/${pkg}/-rev/${manifest._rev}`).reply(201)
+  registry.unpublish({ manifest })
 
   await npm.exec('unpublish', ['test-package'])
   t.equal(joinedOutput(), '- test-package')
@@ -361,7 +361,7 @@ t.test('publishConfig no spec', async t => {
   })
   const manifest = registry.manifest({ name: pkg })
   await registry.package({ manifest, query: { write: true } })
-  registry.nock.delete(`/${pkg}/-rev/${manifest._rev}`).reply(201)
+  registry.unpublish({ manifest })
   await npm.exec('unpublish', [])
   t.equal(joinedOutput(), '- test-package@1.0.0')
 })
@@ -391,9 +391,39 @@ t.test('publishConfig with spec', async t => {
   })
   const manifest = registry.manifest({ name: pkg })
   await registry.package({ manifest, query: { write: true }, times: 2 })
-  registry.nock.delete(`/${pkg}/-rev/${manifest._rev}`).reply(201)
+  registry.unpublish({ manifest })
   await npm.exec('unpublish', ['test-package'])
   t.equal(joinedOutput(), '- test-package')
+})
+
+t.test('scoped registry config', async t => {
+  const scopedPkg = `@npm/test-package`
+  const alternateRegistry = 'https://other.registry.npmjs.org'
+  const { npm } = await loadMockNpm(t, {
+    config: {
+      force: true,
+      '@npm:registry': alternateRegistry,
+      '//other.registry.npmjs.org/:_authToken': 'test-other-token',
+    },
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: pkg,
+        version: '1.0.0',
+        publishConfig: {
+          registry: alternateRegistry,
+        },
+      }, null, 2),
+    },
+  })
+  const registry = new MockRegistry({
+    tap: t,
+    registry: alternateRegistry,
+    authorization: 'test-other-token',
+  })
+  const manifest = registry.manifest({ name: scopedPkg })
+  await registry.package({ manifest, query: { write: true } })
+  registry.unpublish({ manifest })
+  await npm.exec('unpublish', [scopedPkg])
 })
 
 t.test('completion', async t => {

